marray/__init__.py

```diff
@@ -262,13 +262,20 @@ def fun(x, /, *args, name=name, **kwargs):
         setattr(mod, name, fun)
 
     ## Data Type Functions and Data Types ##
-    dtype_fun_names = ['can_cast', 'finfo', 'iinfo', 'isdtype', 'result_type']
+    dtype_fun_names = ['can_cast', 'finfo', 'iinfo', 'result_type']
+    for name in dtype_fun_names:
+        def fun(*args, name=name, **kwargs):
+            args = [(getattr(arg, 'data') if hasattr(arg, 'mask') else arg)
```

Comment:
I know this looks weird, but it's weird in the same way in a few places. Feel free to open an issue, but rather than change this here, I will add a canonical way to test whether something is an `MArray`[^1] and use it where needed.

[^1]: or NumPy masked array (because currently many functions also work for masked arrays, and I don't mind that)